### PR TITLE
add sets.pop procedure (analogue to python)

### PR DESCRIPTION
**Pull request by skilchen**
```nim
proc pop*[A](s: var HashSet[A]): A =
  ## Remove and return an arbitrary element from the set.
  ## Raises KeyError if the set is empty.
  ##
  var elem: A
  for h in 0..high(s.data):
    if isFilled(s.data[h].hcode):
      elem = s.data[h].key
      excl(s, elem)
      return elem
  raise newException(KeyError, "set is empty")
```

### CODE REVIEW

File: lib/pure/collections/sets.nim [hunk 1]
```diff
@@ -347,6 +347,18 @@ proc excl*[A](s: var HashSet[A], other: HashSet[A]) =
...
   for item in other: discard exclImpl(s, item)
 
+proc pop*[A](s: var HashSet[A]): A =
+  ## Remove and return an arbitrary element from the set.
+  ## Raises KeyError if the set is empty.
+  ##
+  var elem: A
+  for h in 0..high(s.data):
+    if isFilled(s.data[h].hcode):
+      elem = s.data[h].key
```
Comment (Araq) [inline lib/pure/collections/sets.nim]: Use ``result``  here.

File: lib/pure/collections/sets.nim [hunk 2]
```diff
@@ -347,6 +347,18 @@ proc excl*[A](s: var HashSet[A], other: HashSet[A]) =
   assert other.isValid, "The set `other` needs to be initialized."
   for item in other: discard exclImpl(s, item)
 
+proc pop*[A](s: var HashSet[A]): A =
+  ## Remove and return an arbitrary element from the set.
+  ## Raises KeyError if the set is empty.
+  ##
+  var elem: A
```
Comment (Araq) [inline lib/pure/collections/sets.nim]: use ``result`` instead of ``elem``.

**Comment by Araq** [comment]
Apart form my nitpicking, it's fine.

File: lib/pure/collections/sets.nim [hunk 3]
```diff
@@ -347,6 +347,17 @@ proc excl*[A](s: var HashSet[A], other: HashSet[A]) =
   assert other.isValid, "The set `other` needs to be initialized."
   for item in other: discard exclImpl(s, item)
 
+proc pop*[A](s: var HashSet[A]): A =
+  ## Remove and return an arbitrary element from the set.
+  ## Raises KeyError if the set is empty.
```
Comment (dom96) [inline lib/pure/collections/sets.nim]: Small nitpick: Next time please add an empty line in between these. They shouldn't be in the same paragraph.

**Pull request merged**